### PR TITLE
[SPARK-28542][DOCS][WebUI] Stages Tab

### PR DESCRIPTION
**Pull request by planga82**
### What changes were proposed in this pull request?
New documentation to explain in detail Web UI Stages page. New images are included to better explanation.
image
image
image
image

### Does this PR introduce any user-facing change?
Only documentation

### How was this patch tested?
I have generated it using "jekyll build" to ensure that it's ok

### CODE REVIEW

**Review by srowen — CHANGES_REQUESTED** [review]

File: docs/web-ui.md [hunk 1]
```diff
@@ -94,9 +94,76 @@ This page displays the details of a specific job identified by its job ID.
...
+  <img src="img/AllStagesPageDetail5.png" title="Stage DAG" alt="Stage DAG" width="50%">
+</p>
+
+Summary metrics for all task are represented in a table and in a timeline
+* **[Tasks deserialization time](configuration.html#compression-and-serialization)**
+* **Duration of tasks**
+* **GC time**
+* **Result serialization time** is the time spent serializing the task result on a executor before sending it back to the driver
+* **Getting result time** is the time that the driver spends fetching task results from workers
+* **Scheduler delay** includes the time to ship the task from the scheduler to executors, and the time to send the task result from the executors to the scheduler
```
[the remark this answers is not in the document]
Comment (planga82) [reply docs/web-ui.md]: Yes, I think it includes return time but it's not true.

File: docs/web-ui.md [hunk 2]
```diff
@@ -94,9 +94,76 @@ This page displays the details of a specific job identified by its job ID.
...
+* **Peak execution memory** is the sum of the peak sizes of the internal data structures created during shuffles, aggregations and joins.
+* **Shuffle Read Size / Records**
+* **Shuffle Read Blocked Time** is the time that tasks spent blocked waiting for shuffle data to be read from remote machines
+* **Shuffle Remote Reads** is the total shuffle bytes read from remote executors
```
[the remark this answers is not in the document]
Comment (planga82) [reply docs/web-ui.md]: I have updated the description. Shuffle Read Size include data read locally and from the remote executors. Shuffle Remote Reads include only data readed from remote executors.
I'm wrong?

**Comment by planga82** [comment]
Thank you very much @srowen for all your comments. I have fixed all comments. I have tried to explain only not obvious things to keep the documentation as simple as possible.

File: docs/web-ui.md [hunk 3]
```diff
@@ -94,9 +94,76 @@ This page displays the details of a specific job identified by its job ID.
...
+</p>
+
+Summary metrics for all task are represented in a table and in a timeline.
+* **[Tasks deserialization time](configuration.html#compression-and-serialization)**
+* **Duration of tasks**.
+* **GC time** is the total JVM garbage collection time.
+* **Result serialization time** is the time spent serializing the task result on a executor before sending it back to the driver.
+* **Getting result time** is the time that the driver spends fetching task results from workers.
+* **Scheduler delay** is the time the task waits to be scheduled for execution.
+* **Peak execution memory** it's the maximum memory used by the internal data structures created during shuffles, aggregations and joins.
```
Comment (srowen) [inline docs/web-ui.md]: it's -> is

File: docs/web-ui.md [hunk 4]
```diff
@@ -94,9 +94,76 @@ This page displays the details of a specific job identified by its job ID.
...
+* **Shuffle Read Size / Records**.Total shuffle bytes read, includes both data read locally and data read from remote executors.
```
Comment (srowen) [inline docs/web-ui.md]: Space before Total

File: docs/web-ui.md [hunk 5]
```diff
@@ -94,9 +94,76 @@ This page displays the details of a specific job identified by its job ID.
...
+* **Shuffle Read Blocked Time** is the time that tasks spent blocked waiting for shuffle data to be read from remote machines.
+* **Shuffle Remote Reads** is the total shuffle bytes read from remote executors.
+* **Shuffle spill (memory)** is the size of the deserialized form of the shuffled data in memory.
+* **Shuffle spill (disk)** is the size of the serialized form of the data on disk.
+
+<p style="text-align: center;">
+  <img src="img/AllStagesPageDetail6.png" title="Stages metrics" alt="Stages metrics">
+</p>
+
+Aggregated metrics by executor show the same information aggregated by executor
```
Comment (srowen) [inline docs/web-ui.md]: Period at the end.

File: docs/web-ui.md [hunk 6]
```diff
@@ -94,9 +94,76 @@ This page displays the details of a specific job identified by its job ID.
...
+
+<p style="text-align: center;">
+  <img src="img/AllStagesPageDetail7.png" title="Stages metrics per executor" alt="Stages metrics per executors">
+</p>
+
+**Accumulators** are a type of shared variables, it provide a mutable variable that can be updated inside of a variety of transformations. It is possible to create accumulators with and without name, but only named accumulators are displayed.
```
Comment (srowen) [inline docs/web-ui.md]: variables, it provide -> variable. It provides
I'd link to the accumulators documentation here.

File: docs/web-ui.md [hunk 7]
```diff
@@ -94,9 +94,76 @@ This page displays the details of a specific job identified by its job ID.
...
+
+<p style="text-align: center;">
+  <img src="img/AllStagesPageDetail8.png" title="Stage accumulator" alt="Stage accumulator">
+</p>
+
+Tasks details show the information of every task
```
Comment (srowen) [inline docs/web-ui.md]: Period at the end. Can we say any more, like the types of information shown? Just a sentence.

**Review by srowen — CHANGES_REQUESTED** [review]

**Comment by planga82** [comment]
Changed!!

**Review by srowen — APPROVED** [review]

**Pull request closed without merging**